### PR TITLE
Login sys

### PR DESCRIPTION
**Pull request by Sam-Bro**
added functionality to registration form and login. registration saves information to database, login uses passport to authenticate. logout button not functional. logged in users trying to go to login/registration will be redirected to members page

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -4,13 +4,20 @@ const bcrypt = require("bcryptjs");
 module.exports = function(sequelize, DataTypes) {
   const User = sequelize.define("User", {
     // The email cannot be null, and must be a proper email before creation
-    first_name: {
```
Comment (wannabewayno) [inline models/User.js:7]: we Need first_name;

talk to Reuben and organise a user_name field to use

File: config/config.js [hunk 2]
```diff
@@ -1,24 +1,24 @@
 require('dotenv').config();
 module.exports = {
   "development": {
-    "username": process.env.MYSQL_USERNAME,
```
Comment (wannabewayno) [inline config/config.js:4]: This isn't secure, we'll be using env variables for testing. 

it's super easy. in the root of our project there's a file called .env.example.

Copy that over to a file called .env and fill out the fields example
MYSQL_USERNAME=root 

server.js loads the environment variables on start up

File: models/User.js (hunk 1 above)
Comment (Sam-Bro) [reply models/User.js:7]: user_name has been removed

File: config/config.js (hunk 2 above)
Comment (Sam-Bro) [reply config/config.js]: reverted back to env variables

**Pull request merged**